docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,286 @@
+
+Find out more about possible file types in the `main NWB documentation <https://nwb-overview.readthedocs.io/en/latest/faq_details/why_hdf5.html#why-use-hdf5-as-the-primary-backend-for-nwb>`_.
+
+Find out more about chunking and compression in the `advanced NWB tutorials for dataset I/O settings <https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/h5dataio.html#sphx-glr-tutorials-advanced-io-h5dataio-py>`_.
+
+Find out more about memory buffering of large source files in the `advanced NWB tutorials for iterative data write <https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/plot_iterative_write.html#sphx-glr-tutorials-advanced-io-plot-iterative-write-py>`_.
+
+
+
+Default configuration
+---------------------
+
+To retrieve a default configuration for an in-memory ``pynwb.NWBFile`` object, use the ``get_default_backend_configuration`` function...
+
+.. code-block:: python
+
+    from datetime import datetime
+    from uuid import uuid4
+
+    from dateutil import tz
+    from neuroconv.tools.nwb_helpers import get_default_backend_configuration
+    from pynwb import NWBFile, TimeSeries
+
+    session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))
+    nwbfile = NWBFile(
+        identifier=str(uuid.uuid4()),
+        session_start_time=session_start_time,
+        session_description="A session of my experiment.",
+    )
+
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    default_backend_configuration = get_default_backend_configuration(
+        nwbfile=nwbfile, backend="hdf5"
+    )
+
+From which a printout of the contents looks like...
+
+.. code-block:: python
+
+    print(default_backend_configuration)
+
+.. code-block:: bash
+
+    Configurable datasets identified using the hdf5 backend
+    -------------------------------------------------------
+
+    acquisition/MyTimeSeries/data
+    -----------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 0.00 GB
+
+      buffer shape : (3,)
+      expected RAM usage : 0.00 GB
+
+      chunk shape : (3,)
+      disk space usage per chunk : 0.00 MB
+
+      compression method : gzip
+
+
+    acquisition/MyTimeSeries/timestamps
+    -----------------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 0.00 GB
+
+      buffer shape : (3,)
+      expected RAM usage : 0.00 GB
+
+      chunk shape : (3,)
+      disk space usage per chunk : 0.00 MB
+
+      compression method : gzip
+
+
+Customization
+-------------
+
+To modify the chunking or buffering patterns and compression method or options, change those values in the ``.dataset_configurations`` object using the location of each dataset as a specifier.
```

Comment:
I am confused mysefl about this. Do compression options have buffering patterns? I thought the buffering was only an iterator concept and the compressors rely only on chunks. am I confused?

Reply:
What is confusing and what rephrasing would make it less confusing?

'the chunking or buffering patterns' *and* 'compression method or options'

Reply:
I'm pretty sure this does adjust buffering and chunking values if changed on something already wrapped with an iterator, though have yet to add automatic wrapper of an iterator as a part of that

They are separate concerns; you specify chunking and buffering separately from compression. Performance and results of compression will indirectly depend on chunking pattern but does not rely on the specification directly

Reply:
Do I understand well that adjusting the buffer_shape will have an effect only if the object is already wrapped in an iterator? So for example, things that we have build with `add_electrical_series`?

Reply:
right, in a follow-up extend to adjust anything sliceable at least